staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_nodes.go

```diff
@@ -0,0 +1,598 @@
+)
+
+type NodeLogsCmdFlags struct {
+	RESTClientGetter genericclioptions.RESTClientGetter
+
+	Path                 string
+	Services             []string
+	Pattern              string
+	PatternCaseSensitive bool
+	SinceTime            string
+	UntilTime            string
+	Boot                 int64
+	Output               string
+	TailLines            int64
+	Role                 string
+	Selector             string
+	Raw                  bool
+	Unify                bool
+
+	InsecureSkipTLSVerifyBackend bool
+
+	genericclioptions.IOStreams
+}
+
+func NewNodeLogsCmdFlags(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *NodeLogsCmdFlags {
+	return &NodeLogsCmdFlags{
+		RESTClientGetter:     restClientGetter,
+		IOStreams:            streams,
+		Path:                 "journal",
+		PatternCaseSensitive: true,
+	}
+}
+
+type NodeLogsCmdOptions struct {
+	Resources []string
+	Selector  string
+	Role      string
+
+	// the log path to fetch
+	Path string
+	// --path=journal specific arguments
+	Pattern              string
+	PatternCaseSensitive bool
+	Boot                 *int64
+	Services             []string
+	SinceTime            *metav1.Time
+	UntilTime            *metav1.Time
+	TailLines            *int64
+	Format               string
+	// output format arguments
+	// raw is set to true when we are viewing the journal and wish to skip prefixing
+	Raw    bool
+	Unify  bool
+	Prefix bool
+
+	InsecureSkipTLSVerifyBackend bool
+
+	Object           runtime.Object
+	GetNodeTimeout   time.Duration
+	RESTClientGetter genericclioptions.RESTClientGetter
+	LogsForObject    polymorphichelpers.LogsForObjectFunc
+	ConsumeRequestFn func(rest.ResponseWrapper, corev1.ObjectReference, io.Writer) error
+
+	genericclioptions.IOStreams
+}
+
+// NewCmdNodeLogs creates a new logs command
+func NewCmdNodeLogs(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	flags := NewNodeLogsCmdFlags(restClientGetter, streams)
+
+	cmd := &cobra.Command{
+		Use:                   logsNodesUsageStr,
+		DisableFlagsInUseLine: true,
+		Short:                 "Display and filter node logs",
+		Long:                  logsNodesLong,
+		Example:               logsNodesExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			o, err := flags.ToOptions(args)
+			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(o.Run())
+		},
+	}
+	flags.AddFlags(cmd)
+	return cmd
+}
+
+// AddFlags registers flags for a cli.
+func (flags *NodeLogsCmdFlags) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&flags.Path, "path", flags.Path, "Retrieve the specified path within the node's /var/logs/ folder. The 'journal' value will allow querying the journal on supported operating systems.")
+	cmd.Flags().StringSliceVarP(&flags.Services, "service", "u", flags.Services, "Return log entries from the specified service(s). Only applies to Linux journal or WinEvent Application provider logs."+
```

Comment:
Would it be less weird to define "service" as "-s" and then add "-u" as a helpful alias to "--service" ?

Reply:
This command is eventually intended to be a subcommand for `kubectl logs` which already has a `-s` flag which collided here. If there's a way to still get this be `-s`, I'd be in favour of that, too.

Reply:
oh.   How about `-S` ?

Reply:
Added a commit to switch to `-S`. Will be squashed later.